assets/js/src/core/modules/data-object/editor/toolbar/language-selection/language-selection.tsx

```diff
@@ -15,10 +15,15 @@ import { useLanguageSelection } from './provider/use-language-selection'
 
 export const LanguageSelection = (): React.JSX.Element => {
   const settings = useSettings()
-  const { currentLanguage, setCurrentLanguage } = useLanguageSelection()
+  const { currentLanguage, setCurrentLanguage, hasLocalizedFields } = useLanguageSelection()
+
+  if (!hasLocalizedFields) {
+    return <></>
+  }
 
   return (
     <BaseLanguageSelection
+      key={ hasLocalizedFields ? 'localized' : 'not-localized' } // Force re-render when hasLocalizedFields changes
```

Comment:
Hm... the `hasLocalizedFields` check cannot have any effect here as it's always true here (see if in line 20)